cura/CuraApplication.py

```diff
@@ -716,12 +741,11 @@ def run(self):
         self.setMainQml(Resources.getPath(self.ResourceTypes.QmlFiles, "Cura.qml"))
         self._qml_import_paths.append(Resources.getPath(self.ResourceTypes.QmlFiles))
 
-        run_headless = self.getCommandLineOption("headless", False)
-        if not run_headless:
+        run_without_gui = self.getCommandLineOption("headless", False) or self.getCommandLineOption("invisible", False)
+        if not run_without_gui:
             self.initializeEngine()
-            controller.setActiveStage("PrepareStage")
```

Comment:
Why did this line get removed? It's needed to set Cura to the correct stage on start.

Reply:
Has been moved to Uranium, but you probably already noticed.

Reply:
I have. I moved it back though as Uranium doesn't know about a "PrepareStage", that's Cura specific.